src/IO/WriteBufferFromFileDecorator.cpp

```diff
@@ -31,11 +39,18 @@ WriteBufferFromFileDecorator::~WriteBufferFromFileDecorator()
     {
         tryLogCurrentException(__PRETTY_FUNCTION__);
     }
+
+    swap(*impl);
 }
 
 void WriteBufferFromFileDecorator::sync()
 {
-    impl->sync();
+    next();
```

Comment:
That line costs me several days of debugging.

WriteIndirectBufferFromRemoteFS::count() is incorrect if impl->next called outside of WriteIndirectBufferFromRemoteFS::next(). That leads to the situation when file size is set to 0 in the file system for file format_version.txt.

Reply:
It is interesting fact, why old WriteBufferFromFileDecorator::sync didn't couse any fails in test?
Because old WriteBufferFromFileDecorator::sync did nothing at all. impl->sync has been called with nulls inside working_buffer and pos du to missing swaps. When I added SwapHelper, impl->sync has began do the work, as a result WriteBufferFromFileDecorator::bytes and Impl.bytes have diverged.